### DISCUSSION TITLE
Distinguishing HEIF and AVIF on load

### DISCUSSION THREAD

**Discussion by kstanikviacbs** [libvips/libvips#3349] [Q&A] [answered]
Hi there! Kudos for great job goes to all maintainers as always :) 

I've come across a small challenge recently which is how to distinguish loaded image format types if both utilise `heifload` like `heif` and `avif`. I've been using `VIPS_META_LOADER` as advised in the comment for that purpose so far, but in this case it's no longer unique so wondering how to achieve that. Do you have any ideas? I would appreciate any help in this matter. 

Thanks,
Konrad

**Comment by lovell** [comment, marked as answer]
Did you see the `heif-compression` metadata? This will be either `av1` or `hevc` for HEIF input.

**Reply by kstanikviacbs** [reply]
I was not aware of it being exposed as metadata as well - that's exactly what I need, thanks @lovell !
